infra/base-images/base-clang/Dockerfile

```diff
@@ -56,6 +55,11 @@ ENV CCC "clang++"
 # FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION is described at
 # https://llvm.org/docs/LibFuzzer.html#fuzzer-friendly-build-mode
 
-ENV CFLAGS "-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
+# The implicit-function-declaration and implicit-int errors are downgraded to a
+# warning, to allow compiling legacy code.
+# See https://releases.llvm.org/16.0.0/tools/clang/docs/ReleaseNotes.html#potentially-breaking-changes
+# Same for vla-cxx-extension, deprecated-declarations
+
+ENV CFLAGS "-O1 -fno-omit-frame-pointer -gline-tables-only -Wno-error=deprecated-declarations -Wno-error=vla-cxx-extension -Wno-error=implicit-function-declaration -Wno-error=implicit-int -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
```

Comment:
> The following remaining projects turn warnings into errors, and cause themselves to fail to build with clang-18:
> <...>
> If the goal is to be developer friendly and aggressively try to compile any code (even if it is not fully conformant) into a fuzz binary, an easier way would be to globally set `no-error` here (for all errors).

Note that the last flag wins (https://godbolt.org/z/64oqzYfoE),
so that won't quite work.

Reply:
> > Note that the last flag wins
> 
> Yes, I am aware.
> 
> > so that won't quite work.
> 
> It works for all projects that do not append error flags, which seems to be all (or at least the majority) of the projects for now.

I'm just saying that it will generally only work for projects that don't adjust warning flags themselves.

Reply:
I don't have a clear goal in mind, just that widely breaking projects is bad but smaller breakages or locally fixing a few projects is OK. So if it will cause wide breakage, we should avoid it.
:-)